cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,192 @@
+from contentstore.course_group_config import GroupConfiguration
+from contentstore.models import MigrateVerifiedTrackCohortsSetting
+from django.conf import settings
+from course_modes.models import CourseMode
+from django.core.management.base import BaseCommand, CommandError
+
+from openedx.core.djangoapps.course_groups.cohorts import CourseCohort
+from openedx.core.djangoapps.course_groups.models import (CourseUserGroup, CourseUserGroupPartitionGroup)
+from openedx.core.djangoapps.verified_track_content.models import VerifiedTrackCohortedCourse
+
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.partitions.partitions import ENROLLMENT_TRACK_PARTITION_ID
+from xmodule.partitions.partitions_service import PartitionService
+
+
+class Command(BaseCommand):
+    """
+    Migrates a course's xblock's group_access from Verified Track Cohorts to Enrollment Tracks
+    """
+
+    def handle(self, *args, **options):
+
+        errors = []
+
+        module_store = modulestore()
+
+        verified_track_cohorts_settings = self._enabled_settings()
+
+        for verified_track_cohorts_setting in verified_track_cohorts_settings:
+            old_course_key = verified_track_cohorts_setting.old_course_key
+            rerun_course_key = verified_track_cohorts_setting.rerun_course_key
+            audit_cohort_names = verified_track_cohorts_setting.get_audit_cohort_names()
+
+            # Get the CourseUserGroup IDs for the audit course names from the old course
+            audit_course_user_group_ids = CourseUserGroup.objects.filter(
+                name__in=audit_cohort_names,
+                course_id=old_course_key,
+                group_type=CourseUserGroup.COHORT,
+            ).values_list('id', flat=True)
+
+            # Get all of the audit CourseCohorts from the above IDs that are RANDOM
+            random_audit_course_user_group_ids = CourseCohort.objects.filter(
+                course_user_group_id__in=audit_course_user_group_ids,
+                assignment_type=CourseCohort.RANDOM
+            ).values_list('course_user_group_id', flat=True)
+
+            # Get the CourseUserGroupPartitionGroup for the above IDs, these contain the partition IDs and group IDs
+            # that are set for group_access inside of modulestore
+            random_audit_course_user_group_partition_groups = list(CourseUserGroupPartitionGroup.objects.filter(
+                course_user_group_id__in=random_audit_course_user_group_ids
+            ))
+
+            if not random_audit_course_user_group_partition_groups:
+                errors.append('No Audit Course Groups found with names: %s' % audit_cohort_names)
```

Comment:
In my course https://jlajoie.sandbox.edx.org/courses/course-v1:christina+1+1, I have a cohort named "foo" that is random (automatic) and mapped to a content group. However, when I run the migration script I am hitting this error: CommandError: No Audit Course Groups found with names: [u'foo']. I don't understand why.